finality-aleph/src/sync/service.rs

```diff
@@ -293,7 +294,9 @@ where
         );
         match self.handler.handle_request(request) {
             Ok(Action::Response(response_items)) => {
-                self.send_to(NetworkData::RequestResponse(response_items), peer);
+                for chunk in MsgLimiter::new(&response_items) {
+                    self.send_to(NetworkData::RequestResponse(chunk.to_vec()), peer.clone());
```

Comment:
Does it correctly handle lost parts and requests them when needed? How does sub-part verification look like? i.e. can a malicious node spam the receiver by sending messages claiming to be a subpart of a bigger message?

Reply:
not really applicable here. We just chunk large msg into smaller ones where the prefix is understandable by the receiver. The spam here would behave exactly the same as without it. We also do not need the sub-part verification

Reply:
the idea is that we hope the chunks will be delivered in order - if not, the receiver will process as many as possible, and then send another request

Reply:
the message we're talking about is a vector of "items", and we verify each item separately

e.g. if there were 5 chunks 100 items each, and the third chunk gets lost, we then request the missing 300 items, because we were able to import 2x100 items

we do not check if a message was requested, we simply process it

we also don't have any spam protection right now